service/dap/server_test.go

```diff
@@ -4805,15 +4804,16 @@ func TestBadInitializeRequest(t *testing.T) {
 
 func TestBadlyFormattedMessageToServer(t *testing.T) {
 	runTest(t, "increment", func(client *daptest.Client, fixture protest.Fixture) {
-		// Send a badly formatted message to the server, and expect it to close the
```

Comment:
Let's not take out a valid test case since there are still cases when we close the connection. You just need a different way of triggering it from the client (unmarshall error?), but the rest of the logic should still stand. In some cases you don't die (new test case) and in some cases you do (old test case modified).

Reply:
Ok added a test case back in.